Shutdown update-agent cleanly, it panics on reboot

When the host is shutting down, `update-agent` will panic. This isn't a huge priority since the host is being shutdown anyway, but users will see it in the logs and worry.

```
0622 20:28:31.776431       1 agent.go:68] Updating status
I0622 20:28:31.776467       1 agent.go:78] Indicating a reboot is needed
I0622 20:28:45.169911       1 agent.go:163] Setting annotations map[string]string{"container-linux-update.v1.coreos.com/reboot-in-progress":"true"}
I0622 20:28:45.190260       1 agent.go:175] Marking node as unschedulable
I0622 20:28:45.208388       1 agent.go:180] Getting pod list for deletion
I0622 20:28:45.228146       1 agent.go:189] Deleting 0 pods
I0622 20:28:45.228252       1 agent.go:199] Node drained, rebooting
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x14252c7]

goroutine 31 [running]:
github.com/coreos/container-linux-update-operator/pkg/updateengine.(*Client).ReceiveStatuses(0xc4202337e0, 0xc420321560, 0xc4203215c0)
        /go/src/github.com/coreos/container-linux-update-operator/pkg/updateengine/client.go:83 +0xf7
created by github.com/coreos/container-linux-update-operator/pkg/agent.(*Klocksmith).watchUpdateStatus
        /go/src/github.com/coreos/container-linux-update-operator/pkg/agent/agent.go:217 +0x138
```

--- a/pkg/agent/agent.go
+++ b/pkg/agent/agent.go
@@ -142,7 +142,8 @@ func (k *Klocksmith) Run() error {
 	}
 
 	// watch update engine for status updates
-	go k.watchUpdateStatus(k.updateStatusCallback)
+	watchUpdateStatusStop := make(chan struct{})
+	go k.watchUpdateStatus(k.updateStatusCallback, watchUpdateStatusStop)
 
 	// block until constants.AnnotationOkToReboot is set
 	for {
@@ -154,6 +155,9 @@ func (k *Klocksmith) Run() error {
 		}
 		glog.Warningf("error waiting for an ok-to-reboot: %v", err)
 	}
+
+	// stop watching the update status by closing the channel
+	close(watchUpdateStatusStop)
 
 	// set constants.AnnotationRebootInProgress and drain self
 	anno = map[string]string{
@@ -207,12 +211,11 @@ func (k *Klocksmith) Run() error {
 	return nil
 }
 
-func (k *Klocksmith) watchUpdateStatus(update func(s updateengine.Status)) {
+func (k *Klocksmith) watchUpdateStatus(update func(s updateengine.Status), stop <-chan struct{}) {
 	glog.Info("Beginning to watch update_engine status")
 
 	oldOperation := ""
 	ch := make(chan updateengine.Status, 1)
-	stop := make(chan struct{})
 
 	go k.ue.ReceiveStatuses(ch, stop)
 

